src/app/qgsmeasuredialog.cpp

```diff
@@ -43,7 +43,20 @@ QgsMeasureDialog::QgsMeasureDialog( QgsMeasureTool* tool, Qt::WindowFlags f )
   mMeasureArea = tool->measureArea();
   mTotal = 0.;
 
+  mUnitsCombo->addItem( tr( "Meters" ) );
+  mUnitsCombo->addItem( tr( "Feet" ) );
+  mUnitsCombo->addItem( tr( "Degrees" ) );
+  mUnitsCombo->addItem( tr( "Nautical Miles" ) );
+
+  QSettings settings;
+  QString units = settings.value( "/qgis/measure/displayunits", "meters" ).toString();
+  mUnitsCombo->setCurrentIndex( mUnitsCombo->findText( units, Qt::MatchFixedString ) );
```

Comment:
UodateSettings gets called each time you make a new line so it will always
reset the combo which is annoying.

On Fri, 2 Jan 2015 11:54 pm Salvatore Larosa <notifications@github.com>
wrote:

> In src/app/qgsmeasuredialog.cpp
> <https://github.com/qgis/QGIS/pull/1747#discussion-diff-22410598>:
>
> > @@ -43,7 +43,20 @@ QgsMeasureDialog::QgsMeasureDialog( QgsMeasureTool* tool, Qt::WindowFlags f )
> >    mMeasureArea = tool->measureArea();
> >    mTotal = 0.;
> >
> > +  mUnitsCombo->addItem( tr( "Meters" ) );
> > +  mUnitsCombo->addItem( tr( "Feet" ) );
> > +  mUnitsCombo->addItem( tr( "Degrees" ) );
> > +  mUnitsCombo->addItem( tr( "Nautical Miles" ) );
> > +
> > +  QSettings settings;
> > +  QString units = settings.value( "/qgis/measure/displayunits", "meters" ).toString();
> > +  mUnitsCombo->setCurrentIndex( mUnitsCombo->findText( units, Qt::MatchFixedString ) );
>
> @NathanW2 <https://github.com/NathanW2> may I suggest to move these two
> lines to updateSettings()?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/1747/files#r22410598>.
>